src/storage/lsm.rs

```diff
@@ -159,6 +160,6 @@ impl LSM {
                 return Ok(v);
             }
         }
-        Ok(Value(None))
+        Ok(Value(OptDatum::Tombstone))
```

Comment:
It the value doesn't exist in the LSM tree, would it make more sense to return `Ok(None)`?

Such that the signature on `get` is:

``` rust
    pub fn get(&self, k: Key) -> Result<Option<Value>>
```

and `Value` is defined as:

``` rust
pub struct Value(pub Datum)
```

and deletions are exposed via

``` rust
fn delete(k: &Key) -> Result<()>
```